dashboard/src/auth/Auth.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import api from '../api/api.js';
+import Login from './Login.js';
+import Loading from '../util/Loading.js';
+import LogoutContext from './LogoutContext.js';
+import { getSessionItem, setSessionItem } from "../util/Storage.js";
+
+const tokenSessionKey = "auth-token";
+
+class Auth extends Component {
+  state = {
+    authenticated: false,
+    showLoading: true,
+    token: getSessionItem(tokenSessionKey) || ""
+  };
+
+  async componentDidMount() {
+    try {
+      api.token = this.state.token;
+      await api.get('/api/operators');
+      this.setState({
+        authenticated: true,
+        showLoading: false,
+        token: api.token
+      });
```

Comment:
If I understand this right, you're setting the global `api` instance's `token` property to the one from state, then doing a call and discarding the result, then setting the state's `token` to that of the `api`.

Does the `token` change during the request? Does the request have side-effects? I'm guessing you only use the request to determine whether you're authenticated and authorized.

Reply:
I'm not really happy with the `Auth` component and the `LogoutContext` but I can't think of an easy way to improve on this either, so lgtm :man_shrugging:

Reply:
correct.